views/js/test/previewer/proxy/item/test.js

```diff
@@ -0,0 +1,731 @@
+], function($, _, urlUtil, communicatorFactory, proxyFactory, itemProxy) {
+    'use strict';
+
+    QUnit.module('itemProxy');
+
+
+    // prevent the AJAX mocks to pollute the logs
+    $.mockjaxSettings.logger = null;
+    $.mockjaxSettings.responseTime = 1;
+
+    // restore AJAX method after each test
+    QUnit.testDone(function () {
+        $.mockjax.clear();
+    });
+
+
+    QUnit.test('module', function(assert) {
+        QUnit.expect(6);
+        assert.equal(typeof itemProxy, 'object', "The itemProxy module exposes an object");
+        assert.equal(typeof proxyFactory, 'function', "The proxyFactory module exposes a function");
+        assert.equal(typeof proxyFactory.registerProvider, 'function', "The proxyFactory module exposes a registerProvider method");
+        assert.equal(typeof proxyFactory.getProvider, 'function', "The proxyFactory module exposes a getProvider method");
+
+        proxyFactory.registerProvider('itemProxy', itemProxy);
+
+        assert.equal(typeof proxyFactory('itemProxy'), 'object', "The proxyFactory factory has registered the itemProxy definition and produces an instance");
+        assert.notStrictEqual(proxyFactory('itemProxy'), proxyFactory('itemProxy'), "The proxyFactory factory provides a different instance of itemProxy on each call");
+    });
+
+
+    QUnit
+        .cases([
+            { title : 'install' },
+            { title : 'init' },
+            { title : 'destroy' },
+            { title : 'callTestAction' },
+            { title : 'getItem' },
+            { title : 'submitItem' },
+            { title : 'callItemAction' }
+        ])
+        .test('proxy API ', function(data, assert) {
+            QUnit.expect(1);
+            assert.equal(typeof itemProxy[data.title], 'function', 'The itemProxy definition exposes a "' + data.title + '" function');
+        });
+
+
+    QUnit
+        .cases([{
+            title: 'success',
+            token: '1234',
+            response: {
+                token: '4567',
+                success: true
+            },
+            ajaxSuccess: true,
+            success: true
+        }, {
+            title: 'failing data',
+            token: '1234',
+            response: {
+                token: '4567',
+                success: false
+            },
+            ajaxSuccess: true,
+            success: false
+        }, {
+            title: 'failing request',
+            token: '1234',
+            response: "error",
+            ajaxSuccess: false,
+            success: false
+        }])
+        .asyncTest('itemProxy.init ', function(caseData, assert) {
+            var initConfig = {
+                serviceCallId: 'foo',
+                bootstrap: {
+                    serviceController: 'MockRunner',
+                    serviceExtension: 'MockExtension'
+                }
+            };
+
+            var expectedUrl = urlUtil.route('init', initConfig.bootstrap.serviceController, initConfig.bootstrap.serviceExtension, {
```

Comment:
Indeed, if the url is called while the instance is in a subfolder it could be a problem.
However, this url is only built to be compared with the one provided by the proxy. It is not intended to be called. Hence the use of mockJax to fake the true requests.